ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -340,10 +303,15 @@ def create_random_action(self, n_agents: int) -> np.ndarray:
                         size=(n_agents),
                         dtype=np.int32,
                     )
-                    for i in range(self.action_size)
+                    for i in range(self.discrete_action_size)
                 ]
             )
-            return action
+        return action
+
+
+class BehaviorSpec(NamedTuple):
+    observation_shapes: List[Tuple]
+    action_spec: ActionSpec
```

Comment:
I think that since this object is already called BehaviorSpec, the property `action_spec` can be simply `action`

env.behavior_specs[name].action seems better than env.behavior_specs[name].action_spec

Reply:
I tend to disagree. I think it should be called action_spec to indicate that it itself contains 'specs' (and maybe utilities).

Reply:
I'm in favor of `action_spec` too, since it's an `ActionSpec`